Factory/HttpFoundationFactory.php

```diff
@@ -104,32 +104,21 @@ private function getFiles(array $uploadedFiles)
     private function createUploadedFile(UploadedFileInterface $psrUploadedFile)
     {
         $temporaryPath = '';
-        $clientFileName = '';
-        if (UPLOAD_ERR_NO_FILE !== $psrUploadedFile->getError()) {
+        $clientFileName = $psrUploadedFile->getClientFilename();
+        $stream = $psrUploadedFile->getStream();
+        $temporaryPath = $stream->getMetadata('uri');
+
+        if ($test = !\is_string($temporaryPath) || !is_uploaded_file($temporaryPath)) {
             $temporaryPath = $this->getTemporaryPath();
             $psrUploadedFile->moveTo($temporaryPath);
-
-            $clientFileName = $psrUploadedFile->getClientFilename();
-        }
-
-        if (class_exists('Symfony\Component\HttpFoundation\HeaderUtils')) {
```

Comment:
This looks like a different cleanup, not something related to the `moveTo` refactoring. It should be submitted as a dedicated commit (and probably even PR)